src/com/projectkorra/projectkorra/ability/CoreAbility.java

```diff
@@ -383,7 +383,7 @@ public static CoreAbility getAbility(final Class<? extends CoreAbility> clazz) {
 	 * @param clazz the class for the type of CoreAbilities
 	 * @return a Collection of real instances
 	 */
-	public static <T extends CoreAbility> Collection<T> getAbilities(final Player player, final Class<T> clazz) {
+	public static <T extends CoreAbility> Collection<T> getAbilities(final Player player, final Class<? extends T> clazz) {
```

Comment:
Why would you add the generic wildcard here which forces you to explicitly downcast later on lines 644 and 713?